src/main/java/com/keniu/services/ShoppingCartServiceImpl.java

```diff
@@ -0,0 +1,85 @@
+package com.keniu.services;
+
+import com.keniu.dto.CreateCartItemRequestDto;
+import com.keniu.dto.ShoppingCartDto;
+import com.keniu.dto.UpdateCarItemRequestDto;
+import com.keniu.exceptions.EntityNotFoundException;
+import com.keniu.mappers.ShoppingCartMapper;
+import com.keniu.models.Book;
+import com.keniu.models.CartItem;
+import com.keniu.models.ShoppingCart;
+import com.keniu.repositories.ShoppingCartRepository;
+import jakarta.transaction.Transactional;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class ShoppingCartServiceImpl implements ShoppingCartService {
+    private final ShoppingCartRepository shoppingCartRepository;
+    private final ShoppingCartMapper shoppingCartMapper;
+
+    @Override
+    @Transactional
```

Comment:
Do you need `@Transactional` here?

Reply:
https://github.com/kenu21/book-shop/pull/10#discussion_r1860468930

Reply:
Please coordinate with the mentors on what needs to be done and how. I’m not willing to redo the same thing over and over just because everyone of us has a different opinion